worlds/blasphemous/Preprocessor.py

```diff
@@ -0,0 +1,579 @@
+                            parens_conditions.append(build_logic_subconditions(current_condition, sub_part))
+                            #print("PARENS:", parens_conditions)
+
+                            temp_conditions: List[List[str]] = []
+
+                            for i in parens_conditions[0]:
+                                for j in parens_conditions[1]:
+                                    temp_conditions.append(i + j)
+
+                            parens_conditions.pop(0)
+                            parens_conditions.pop(0)
+
+                            while len(parens_conditions) > 0:
+                                temp_conditions2 = temp_conditions
+                                temp_conditions = []
+                                for k in temp_conditions2:
+                                    for l in parens_conditions[0]:
+                                        temp_conditions.append(k + l)
+                                
+                                parens_conditions.pop(0)
+
+                            #print("TEMP:", remove_duplicates(temp_conditions))
+                            all_conditions += temp_conditions
+                        else:
+                            all_conditions += build_logic_subconditions(current_condition, sub_part)
+
+                    current_index = index+2
+                    
+                    current_condition = []
+                    sub_part = ""
+                    
+            continue
+
+        # collect all parts until reaching end of parentheses
+        if parens > 0:
+            sub_part += f" {part}"
+            continue
+
+        current_condition.append(part)
+
+        # continue with current condition
+        if parts[index+1] == "&&":
+            current_index = index+2
+            continue
+        
+        # add condition to list and start new one
+        elif parts[index+1] == "||":
+            if len(parens_conditions) > 0:
+                for i in parens_conditions:
+                    for j in i:
+                        all_conditions.append(j + current_condition)
+                parens_conditions = []
+            else:    
+                all_conditions.append(current_condition)
+            current_condition = []
+            current_index = index+2
+            continue
+        
+    return remove_duplicates(all_conditions)
+
+
+def build_logic_subconditions(current_condition: List[str], subcondition: str) -> List[List[str]]:
+    #print("STARTED SUBCONDITION", current_condition, subcondition)
+    subconditions = build_logic_conditions(subcondition[1:-1])
+    final_conditions = []
+
+    for condition in subconditions:
+        final_condition = current_condition + condition
+        final_conditions.append(final_condition)
+
+    #print("ENDED SUBCONDITION")
+    #print(final_conditions)
+    return final_conditions
+
+
+def remove_duplicates(conditions: List[List[str]]) -> List[List[str]]:
```

Comment:
You can avoid this by using sets instead of lists, though it's possible you may want to attempt to remove proper supersets (maybe this could come up with relation to door visibility somehow)

Reply:
lists are not hashable, so they can't go inside sets

Witness uses FrozenSet[FrozenSet[str]] to get around this.

Reply:
Also REMOVE DUPLICATES??? I LITERALLY HAVE THIS EXACT FUNCTION I'm losing my mind y'all are doing what I'm doing and I didn't realize

Reply:
https://github.com/ArchipelagoMW/Archipelago/blob/main/worlds/witness/data/utils.py#L236-L247

Reply:
My version removes supersets as well